bluetooth/update/README.md

```diff
@@ -2,145 +2,133 @@
+3. Join the detector's geometry to the closest centreline intersection
+4. Route the segments using street centreline's intersection `gis.centreline_both_dir` using [pgr_dijkstra].
 
-```SQL
-WITH from_join AS (
-SELECT * FROM rliu.new_bt_corridors B
-CROSS JOIN rliu.new_bt_readers A
-WHERE levenshtein(A.reader_name, B.from_intersection, 1, 1, 2) < 4)
-```
 
-Matches the corridors to the names based on similar from intersections.
+This .csv file was then imported into QGIS and pushed into the mohan schema in postgres. The geometry column is created by adding a column using `ALTER TABLE mohan.new_added_detectors ADD COLUMN from_geom GEOMETRY` we need two geometry columns: `from_geom` and `to_geom`.  Filling in the from_geom column with `UPDATE mohan.new_added_detectors SET to_geom = ST_SetSRID(ST_MakePoint(to_lon, to_lat), 4326)` and modify the query for `from_geom`.
 
-```SQL
-, to_join AS (
-SELECT * FROM rliu.new_bt_corridors B
-CROSS JOIN rliu.new_bt_readers A
-WHERE levenshtein(A.reader_name, B.to_intersection, 1, 1, 2) < 4)
-```
-Does the same thing for the to_intersections
-
-```SQL
-, joined_readers AS (
-SELECT A.analysis_id, A.corridor, A.reader_name AS from_intersection, A.long AS from_long, A.lat AS from_lat, B.reader_name AS to_intersection, B.long AS to_long, B.lat AS to_lat FROM from_join A
-INNER JOIN to_join B USING (analysis_id)
-)
-```
+## Finding Nearest Intersection IDs
 
-Makes a single table for the corridors and readers at the from/to intersections.
+To get the intersection ids that are close to the newly added detectors location, create a table named `bluetooth_nodes`. This table has four fields:
+`bluetooth_id`, `geom` (this is geometry of bluetooth detectors), `int_id` (nearest intersection id) and `int_geom` (geometry of the nearest intersection to the )
 
+This table is created using the following query:
 ```SQL
-
-, make_lines AS (
-SELECT analysis_id, 
-	corridor, 
-	from_intersection, 
-	from_long, 
-	from_lat, 
-	to_intersection, 
-	to_long, 
-	to_lat,
-	ST_MakeLine(ST_MakePoint(from_long, from_lat), ST_MakePoint(to_long, to_lat)) AS line,
-	ST_LENGTH(ST_MakeLine(ST_MakePoint(from_long, from_lat), ST_MakePoint(to_long, to_lat))) AS length 
-FROM 	joined_readers
---Since there are two readers named Sheppard, this is a check to stop it from matching to the wrong reader
-WHERE  ST_LENGTH(ST_MakeLine(ST_MakePoint(from_long, from_lat), ST_MakePoint(to_long, to_lat)))<0.1
-ORDER BY from_intersection
-)
+CREATE TABLE mohan.bluetooth_nodes AS(
+SELECT DISTINCT mohan.new_added_detectors.from_id::integer AS bluetooth_id,
+    mohan.new_added_detectors.from_geom,
+    nodes.int_id,
+    st_transform(nodes.node_geom, 4326) AS int_geom
+   FROM mohan.new_added_detectors
+     CROSS JOIN LATERAL ( SELECT z.int_id,
+            st_transform(z.geom, 98012) AS node_geom
+           FROM gis.centreline_intersection z
+          ORDER BY (z.geom <-> mohan.new_added_detectors.from_geom)
```

Comment:
would we miss detectors because only from detector's geometry was used?

Reply:
feel like we should be using a distinct detector id of both from and to detector